### ISSUE TITLE
failing without clear reason

### ISSUE THREAD

**Issue by sibelius** [node-fetch/node-fetch#1786]
**Reproduction**

it throws this error sometimes when accessing: https://openpix.com.br/pay/84c1b233-855c-4590-9e41-16734290c6a2

```jsx
FetchError: request to https://openpix.com.br/pay/graphql failed, reason: 
    at ClientRequest.<anonymous> (webpack://index/node_modules/node-fetch/lib/index.js:1491:11)
    at ClientRequest.emit (node:events:523:35)
    at ClientRequest.emit (node:domain:489:12)
    at TLSSocket.socketErrorListener (node:_http_client:495:9)
    at TLSSocket.emit (node:events:511:28)
    at TLSSocket.emit (node:domain:489:12)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

The related to is here

```jsx
req.on('error', function (err) {
			reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));

			if (response && response.body) {
				destroyStream(response.body, err);
			}

			finalize();
		});
```

Try to access this https://openpix.com.br/pay/84c1b233-855c-4590-9e41-16734290c6a2 and reload or use a anonymous navigator to try to trigger this error

**Expected behavior**

It should not error

**Screenshots**

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       | 2.7.0
| node             | 20-alpine
| npm              | 10.1.0
| Operating System | linux

**Additional context**

How can I get more info on this bug? why reason is not defined?

**Comment by OliverJAsh** [comment]
I am facing the same issue. We're using `error.message` and the "reason" is empty. Reduced test case:

```ts
import fetch from 'node-fetch';
fetch('http://localhost:1234/regdvfs');

```

Result:

```console
file:///Users/oliver/Code/unsplash/unsplash-web/node_modules/node-fetch/src/index.js:108
			reject(new FetchError(`request to ${request.url} failed, reason: ${error.message}`, 'system', error));
			       ^

FetchError: request to http://localhost:1234/regdvfs failed, reason:
    at ClientRequest.<anonymous> (file:///Users/oliver/Code/unsplash/unsplash-web/node_modules/node-fetch/src/index.js:108:11)
    at ClientRequest.emit (node:events:518:28)
    at Socket.socketErrorListener (node:_http_client:500:9)
    at Socket.emit (node:events:518:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  erroredSysCall: undefined
}
```

`error.message`:

```
FetchError: request to http://localhost:1234/regdvfs failed, reason:
```

I'd expect something like this:

```
FetchError: request to http://localhost:1234/regdvfs failed, reason: ECONNREFUSED
```

**Comment by sibelius** [comment]
mine was a failing dns service

**Comment by OliverJAsh** [comment]
Ah, but I think this is still an issue—that `node-fetch` doesn't return a more informative error `message`?

**Comment by ShiiRochi** [comment]
> Ah, but I think this is still an issue—that `node-fetch` doesn't return a more informative error `message`?

I faced similar issue, no proxy, no VPNs, but deployed within docker using node:20-alpine3.17 image, after downgrading to node:18-alpine3.19 issue resolved. Node fetch was causing error while fetching image from CDN.

**Comment by s100** [comment]
I think this is Windows-specific. We see this issue with Node.js 20.17.0 and Node.js 22.10.0 on Windows. We do not see this issue with Node.js 18.20.4 on Windows, and we do not see it with Node.js 18/20/22 on Linux.

**Comment by davecrighton** [comment]
Assuming I am talking about the same occurrence as @s100 I am actually seeing locally on linux but doesn't seem to occur on our Linux CI builds.

**Comment by yotov** [comment]
I found this thread because I experienced a similar problem and I'm just dropping a comment in case it's useful to someone.

In our case, the problem was caused by high latency (300 ms) and the fact that Node.js waits 250 ms by default due to the Happy Eyeballs algorithm before timing out. After disabling the Happy eyeballs ```--no-network-family-autoselection``` we stop to see such errors.

Here are more details: https://r1ch.net/blog/node-v20-aggregateeerror-etimedout-happy-eyeballs
